From: Padmanabh Ratnakar <padmanabh.ratnakar@emulex.com>
Date: Sat, 11 Jun 2011 22:58:46 +0000 (-0700)
Subject: be2net: Enable NETIF_F_TSO6 for VLAN traffic for BE
References: FATE#311448, bnc#697255
Git-repo: git://git.kernel.org/pub/scm/linux/kernel/git/davem/net-next-2.6.git
Git-commit: eb8a50d9a6118a310c0b01d47091a2a5d994b549
Patch-mainline: Probably v3.1
Git-commit: eb8a50d9a6118a310c0b01d47091a2a5d994b549

be2net: Enable NETIF_F_TSO6 for VLAN traffic for BE

NETIF_F_TSO6 for VLAN packets was not enabled for BE adapters.
Enabling it.

Signed-off-by: Padmanabh Ratnakar <padmanabh.ratnakar@emulex.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Acked-by: Jeff Mahoney <jeffm@suse.com>
---

 drivers/net/benet/be_main.c |    5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

--- a/drivers/net/benet/be_main.c
+++ b/drivers/net/benet/be_main.c
@@ -2833,12 +2833,9 @@ static void be_netdev_init(struct net_de
 	netdev->features |= netdev->hw_features |
 		NETIF_F_HW_VLAN_RX | NETIF_F_HW_VLAN_FILTER;
 
-	netdev->vlan_features |= NETIF_F_SG | NETIF_F_TSO |
+	netdev->vlan_features |= NETIF_F_SG | NETIF_F_TSO | NETIF_F_TSO6 |
 		NETIF_F_IP_CSUM | NETIF_F_IPV6_CSUM;
 
-	if (lancer_chip(adapter))
-		netdev->vlan_features |= NETIF_F_TSO6;
-
 	netdev->flags |= IFF_MULTICAST;
 
 	/* Default settings for Rx and Tx flow control */


